tests/test_reauth.py

```diff
@@ -0,0 +1,363 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the reauth module."""
+
+import base64
+import json
+import os
+import unittest
+
+import mock
+
+from six.moves import http_client
+from six.moves import urllib
+
+from oauth2client import client
+
+from google_reauth import challenges
+from google_reauth import reauth
+from google_reauth import errors
+from google_reauth import reauth_creds
+from google_reauth import _reauth_client
+from google_reauth.reauth_creds import Oauth2WithReauthCredentials
+
+from pyu2f import model
+from pyu2f import u2f
+
+
+_ok_response = lambda: None
+setattr(_ok_response, 'status', http_client.OK)
+
+_error_response = lambda: None
+setattr(_error_response, 'status', None)
+
+
+class ReauthTest(unittest.TestCase):
+    """This class contains tests for mocking reauth.
+
+    The tests here are a bit more verbose since we are trying to mock out U2F, and
+    present the reauth code some form of valid reply. This makes this test a bit
+    of an integration test, as opposed to testing every individual method by
+    itself.
+    """
+
+    rapt_token = 'encoded_proof_of_reauth_token'
+    correct_password = 'correct_password'
+    oauth_api_url = 'http://some_url.com'
+    client_id = 'some_id'
+    client_secret = 'some_secret'
+
+    def _request_mock_side_effect(self, *args, **kwargs):
```

Comment:
I suggest taking a look here: https://github.com/GoogleCloudPlatform/google-auth-library-python-httplib2/blob/master/tests/test_google_auth_httplib2.py#L24-L42

Your mock should be configured per-test instead of having one big one. (happy to help deconstruct this if needed)

Reply:
Request mock is configured per test, this method is just a common side effect for a flow with no errors.